src/tls/openssl/tls_tcp.c

```diff
@@ -229,7 +229,13 @@ static bool recv_handler(int *err, struct mbuf *mb, bool *estab, void *arg)
 
 	if (SSL_state(tc->ssl) != SSL_ST_OK) {
 
-		if (tc->up) {
+		if (tc->up
+#if !defined(LIBRESSL_VERSION_NUMBER)
+			&& SSL_state(tc->ssl) != TLS_ST_CW_CLNT_HELLO
+			&& SSL_state(tc->ssl) != TLS_ST_CW_FINISHED
+			&& SSL_state(tc->ssl) != TLS_ST_SW_SRVR_DONE
+#endif
+		) {
```

Comment:
it is a bit difficult to read ...

perhaps it is better to declare a bool before the if and use that bool inside the if ?